File: android/app/src/main/java/dev/bennett/codexmeter/UsageFormat.java
package dev.bennett.codexmeter;

import android.content.Context;
import android.text.format.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

/* JADX INFO: loaded from: classes.dex */
public final class UsageFormat {
    private UsageFormat() {
    }

    public static String planLabel(String str) {
        if (str == null || str.trim().isEmpty()) {
            return "";
        }
        String normalized = str.trim().toLowerCase(Locale.US).replace("_", "").replace("-", "");
        switch (normalized) {
            case "free": return "Free";
            case "go": return "Go";
            case "plus": return "Plus";
            case "prolite":
            case "pro5x": return "Pro 5x";
            case "pro":
            case "pro20x": return "Pro 20x";
            default: return "";
        }
    }

    public static String percent(UsageWindow usageWindow, String str, boolean z) {
        if (usageWindow == null) {
            return z ? "—" : Translations.t("Unavailable");
        }
        boolean zEquals = WidgetOptions.DISPLAY_USED.equals(str);
        int iRemainingPercent = zEquals ? usageWindow.usedPercent : usageWindow.remainingPercent();
        if (z) {
            return iRemainingPercent + "%";
        }
        return iRemainingPercent + "% " + Translations.t(zEquals ? WidgetOptions.DISPLAY_USED : "left");
    }

    public static String reset(Context context, UsageWindow usageWindow, String str, long j) {
        return reset(context, usageWindow, str, j, j);
    }

    public static String reset(Context context, UsageWindow usageWindow, String str,
            long observedAtMillis, long nowMillis) {
        if (usageWindow == null || WidgetOptions.RESET_HIDDEN.equals(str)
                || !usageWindow.showsResetCountdown()) {
            return "";
        }
        long jResetAtMillis = usageWindow.effectiveResetAtMillis(observedAtMillis);
        if (jResetAtMillis <= 0) {
            return Translations.t("Reset time unavailable");
        }
        String strAbsolute = absolute(context, jResetAtMillis, nowMillis);
        String strRelative = relative(jResetAtMillis, nowMillis);
        if (WidgetOptions.RESET_RELATIVE.equals(str)) {
            return Translations.t("Resets ") + strRelative;
        }
        return "both".equals(str) ? Translations.t("Resets ") + strAbsolute + " (" + strRelative + ")" : Translations.t("Resets ") + strAbsolute;
    }

    public static String estimatedRemaining(UsagePace.Assessment assessment) {
        if (assessment == null || !assessment.available) {
            return "";
        }
        if (assessment.estimatedRemainingMillis <= 0L) {
            return Translations.t("Est. depleted");
        }
        return "Est. " + compactDuration(assessment.estimatedRemainingMillis);
    }

    static String compactDuration(long millis) {
        long minutes = Math.max(1L, TimeUnit.MILLISECONDS.toMinutes(Math.max(0L, millis)));
        long days = minutes / 1440L;
        long hours = (minutes % 1440L) / 60L;
        long remainingMinutes = minutes % 60L;
        if (days > 0L) {
            return days + "d " + hours + "h";
        }
        if (hours > 0L) {
            return hours + "h " + remainingMinutes + "m";
        }
        return minutes + "m";
    }

    public static String absolute(Context context, long j, long j2) {
        String str;
        boolean zIs24HourFormat = DateFormat.is24HourFormat(context);
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(j);
        Calendar calendar2 = Calendar.getInstance();
        calendar2.setTimeInMillis(j2);
        Calendar calendar3 = (Calendar) calendar2.clone();
        calendar3.add(6, 1);
        if (sameDay(calendar, calendar2)) {
            str = zIs24HourFormat ? "'today at' HH:mm" : "'today at' h:mm a";
        } else if (sameDay(calendar, calendar3)) {
            str = zIs24HourFormat ? "'tomorrow at' HH:mm" : "'tomorrow at' h:mm a";
        } else {
            str = zIs24HourFormat ? "EEE, MMM d 'at' HH:mm" : "EEE, MMM d 'at' h:mm a";
        }
        if (Locale.getDefault().getLanguage().equals("fr")) {
            str = str.replace("'today at'", "'aujourd’hui à'")
                    .replace("'tomorrow at'", "'demain à'").replace("'at'", "'à'")
                    .replace("EEE, MMM d", "EEE d MMM");
        }
        return new SimpleDateFormat(str, Locale.getDefault()).format(new Date(j));
    }

    private static boolean sameDay(Calendar calendar, Calendar calendar2) {
        return calendar.get(0) == calendar2.get(0) && calendar.get(1) == calendar2.get(1) && calendar.get(6) == calendar2.get(6);
    }

    public static String relative(long j, long j2) {
        boolean french = Locale.getDefault().getLanguage().equals("fr");
        long minutes = TimeUnit.MILLISECONDS.toMinutes(Math.max(0L, j - j2));
        long j3 = minutes / 1440;
        long j4 = (minutes % 1440) / 60;
        long j5 = minutes % 60;
        if (j3 > 0) {
            return (french ? "dans " : "in ") + j3 + (french ? "j " : "d ") + j4 + "h";
        }
        if (j4 > 0) {
            return (french ? "dans " : "in ") + j4 + "h " + j5 + "m";
        }
        return minutes > 0 ? (french ? "dans " : "in ") + minutes + "m" : Translations.t("now");
    }

    public static String updated(long time, long now) {
        return time <= 0 ? Translations.t("Not updated yet")
                : Translations.t("Updated ") + LocalizedTime.relative(time, now, 60000);
    }
}

File: ios/ModelsMeterCore/Sources/ModelsMeterCore/UsageFormat.swift
import Foundation

public enum UsageDisplayMode: String, Codable, Sendable, Equatable, CaseIterable {
    case used
    case remaining
}

public enum UsageResetDisplayMode: String, Codable, Sendable, Equatable, CaseIterable {
    case hidden
    case absolute
    case relative
    case both
}

public enum UsageFormat {
    public static func planLabel(_ planType: String?) -> String {
        guard let planType else {
            return ""
        }
        let normalized = planType
            .trimmingCharacters(in: .whitespacesAndNewlines)
            .lowercased()
            .replacingOccurrences(of: "_", with: "")
            .replacingOccurrences(of: "-", with: "")

        switch normalized {
        case "free": return "Free"
        case "go": return "Go"
        case "plus": return "Plus"
        case "prolite", "pro5x": return "Pro 5x"
        case "pro", "pro20x": return "Pro 20x"
        default: return ""
        }
    }

    public static func percent(
        _ window: UsageWindow?,
        display: UsageDisplayMode = .remaining,
        compact: Bool = false,
        showsPercentSymbol: Bool = true
    ) -> String {
        guard let window else {
            return compact ? "—" : MeterL10n.text("Unavailable", "Indisponible")
        }
        let value = display == .used ? window.usedPercent : window.remainingPercent
        let number = showsPercentSymbol ? "\(value)%" : "\(value)"
        guard !compact else {
            return number
        }
        return "\(number) \(display == .used ? MeterL10n.text("used", "utilisés") : MeterL10n.text("left", "restants"))"
    }

    public static func reset(
        _ window: UsageWindow?,
        display: UsageResetDisplayMode,
        fetchedAt: Date,
        now: Date = Date(),
        calendar: Calendar = .current,
        locale: Locale = MeterL10n.language.locale,
        uses24HourClock: Bool = false
    ) -> String {
        guard let window, display != .hidden, window.showsResetCountdown else {
            return ""
        }
        guard let resetAt = window.effectiveResetDate(relativeTo: fetchedAt) else {
            return MeterL10n.text("Reset time unavailable", "Date de réinitialisation indisponible")
        }

        let absolute = absolute(
            resetAt,
            relativeTo: now,
            calendar: calendar,
            locale: locale,
            uses24HourClock: uses24HourClock
        )
        let relative = relative(until: resetAt, from: now)
        switch display {
        case .hidden:
            return ""
        case .relative:
            return MeterL10n.text("Resets ", "Réinitialisation ") + relative
        case .both:
            return MeterL10n.text("Resets ", "Réinitialisation ") + "\(absolute) (\(relative))"
        case .absolute:
            return MeterL10n.text("Resets ", "Réinitialisation ") + absolute
        }
    }

    public static func absolute(
        _ date: Date,
        relativeTo now: Date,
        calendar: Calendar = .current,
        locale: Locale = MeterL10n.language.locale,
        uses24HourClock: Bool = false
    ) -> String {
        let prefix: String
        if calendar.isDate(date, inSameDayAs: now) {
            prefix = MeterL10n.text("today at", "aujourd’hui à")
        } else if let tomorrow = calendar.date(byAdding: .day, value: 1, to: now),
                  calendar.isDate(date, inSameDayAs: tomorrow) {
            prefix = MeterL10n.text("tomorrow at", "demain à")
        } else {
            let dayFormatter = DateFormatter()
            dayFormatter.calendar = calendar
            dayFormatter.locale = locale
            dayFormatter.dateFormat = "EEE, MMM d"
            prefix = dayFormatter.string(from: date) + MeterL10n.text(" at", " à")
        }

        let timeFormatter = DateFormatter()
        timeFormatter.calendar = calendar
        timeFormatter.locale = locale
        timeFormatter.dateFormat = uses24HourClock ? "HH:mm" : "h:mm a"
        return "\(prefix) \(timeFormatter.string(from: date))"
    }

    public static func relative(until date: Date, from now: Date = Date()) -> String {
        let totalMinutes = Int64(max(0, date.timeIntervalSince(now)) / 60)
        let days = totalMinutes / 1_440
        let hours = (totalMinutes % 1_440) / 60
        let minutes = totalMinutes % 60
        if days > 0 {
            return MeterL10n.text("in \(days)d \(hours)h", "dans \(days) j \(hours) h")
        }
        if hours > 0 {
            return MeterL10n.text("in \(hours)h \(minutes)m", "dans \(hours) h \(minutes) min")
        }
        return totalMinutes > 0 ? MeterL10n.text("in \(totalMinutes)m", "dans \(totalMinutes) min") : MeterL10n.text("now", "maintenant")
    }

    public static func updated(fetchedAt: Date?, now: Date = Date()) -> String {
        guard let fetchedAt else {
            return MeterL10n.text("Not updated yet", "Pas encore actualisé")
        }
        let minutes = Int64(max(0, now.timeIntervalSince(fetchedAt)) / 60)
        if minutes < 1 {
            return MeterL10n.text("Updated just now", "Actualisé à l’instant")
        }
        if minutes < 60 {
            return MeterL10n.text("Updated \(minutes)m ago", "Actualisé il y a \(minutes) min")
        }
        let hours = minutes / 60
        return hours < 24 ? MeterL10n.text("Updated \(hours)h ago", "Actualisé il y a \(hours) h") : MeterL10n.text("Updated \(hours / 24)d ago", "Actualisé il y a \(hours / 24) j")
    }
}
